cppcache/src/statistics/StatisticsTypeImpl.cpp

```diff
@@ -29,32 +29,26 @@ namespace statistics {
 using client::IllegalArgumentException;
 using client::NullPointerException;
 
-StatisticsTypeImpl::StatisticsTypeImpl(std::string nameArg,
-                                       std::string descriptionArg,
-                                       StatisticDescriptor** statsArg,
-                                       int32_t statsLengthArg) {
+StatisticsTypeImpl::StatisticsTypeImpl(
+    std::string nameArg, std::string descriptionArg,
+    std::vector<StatisticDescriptor*> statsArg) {
   if (nameArg.empty()) {
     const char* s = "Cannot have an empty statistics type name";
     throw NullPointerException(s);
   }
-  if (statsArg == nullptr) {
-    const char* s = "Cannot have a null statistic descriptors";
-    throw NullPointerException(s);
-  }
-  if (statsLengthArg > MAX_DESCRIPTORS_PER_TYPE) {
+  if (stats.size() > MAX_DESCRIPTORS_PER_TYPE) {
     throw IllegalArgumentException(
-        "The requested descriptor count " + std::to_string(statsLengthArg) +
+        "The requested descriptor count " + std::to_string(stats.size()) +
         " exceeds the maximum which is " +
         std::to_string(MAX_DESCRIPTORS_PER_TYPE) + ".");
   }
   this->name = nameArg;
   this->description = descriptionArg;
-  this->stats = statsArg;
-  this->statsLength = statsLengthArg;
+  this->stats = std::move(statsArg);
   int32_t intCount = 0;
   int32_t longCount = 0;
   int32_t doubleCount = 0;
-  for (int32_t i = 0; i < this->statsLength; i++) {
+  for (uint32_t i = 0; i < stats.size(); i++) {
```

Comment:
Type of `vector<>::size()` is `size_t`.
Could we not just iterate over the `vector` though and avoid the counter?
This pattern is in a few places.

Reply:
Yes, good idea.